tar/read.c

```diff
@@ -182,7 +184,7 @@ read_archive(struct bsdtar *bsdtar, char mode, struct archive *writer)
 	if (reader_options != NULL) {
 		char *p;
 		/* Set default read options. */
-		p = malloc(sizeof(IGNORE_WRONG_MODULE_NAME)
+		p = (char*)malloc(sizeof(IGNORE_WRONG_MODULE_NAME)
```

Comment:
It was an IntelliSense warning. I thought it was a compiler warning but I was wrong. It still compiles - I'll roll this back out if/when I resubmit the pull request. 

IntelliSense: a value of type "void *" cannot be assigned to an entity of type "char *"	c:\Users\User\Development\libarchive\libarchive\tar\read.c	187	5	bsdtar